packages/sit-onyx/src/components/OnyxUserMenu/future/OnyxUserMenu.vue

```diff
@@ -0,0 +1,173 @@
+<script lang="ts" setup generic="TValue extends SelectOptionValue = SelectOptionValue">
+import chevronLeftSmall from "@sit-onyx/icons/chevron-left-small.svg?raw";
+import { computed } from "vue";
+import { injectI18n } from "../../../i18n";
+import type { SelectOptionValue } from "../../../types";
+import OnyxAvatar from "../../OnyxAvatar/OnyxAvatar.vue";
+import OnyxFlyoutMenu from "../../OnyxFlyoutMenu/future/OnyxFlyoutMenu.vue";
+import OnyxIcon from "../../OnyxIcon/OnyxIcon.vue";
+import OnyxListItem from "../../OnyxListItem/OnyxListItem.vue";
+import type { OnyxUserMenuProps } from "../types";
+
+const props = defineProps<OnyxUserMenuProps<TValue>>();
+
+const emit = defineEmits<{
+  /**
+   * Emitted when an option is clicked.
+   */
+  optionClick: [value: TValue];
+}>();
+
+const slots = defineSlots<{
+  /**
+   * Optional footer content to display at the bottom.
+   */
+  footer?(): unknown;
+}>();
+
+const { t } = injectI18n();
+
+const avatar = computed(() => {
+  if (typeof props.avatar === "object") return { ...props.avatar, label: props.username };
+  return { src: props.avatar, label: props.username };
+});
+</script>
+
+<template>
+  <div class="onyx-user-menu">
+    <OnyxFlyoutMenu class="onyx-user-menu__flyout" :aria-label="t('navigation.userMenuLabel')">
+      <button class="onyx-user-menu__trigger onyx-text">
+        <OnyxAvatar v-bind="avatar" size="24px" />
+        <span class="onyx-truncation-ellipsis"> {{ props.username }}</span>
+        <OnyxIcon class="onyx-user-menu__chevron" :icon="chevronLeftSmall" />
```

Comment:
The chevron does not update correctly when closing the menu by clicking the button:
![image](https://github.com/SchwarzIT/onyx/assets/67898185/74e76005-9791-4451-9390-617186b02b74)